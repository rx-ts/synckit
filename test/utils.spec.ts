import { fileURLToPath } from 'node:url'

import { _dirname } from './helpers'

import { isFile } from 'synckit'

test('utils', () => {
  expect(isFile(_dirname)).toBe(false)
  expect(isFile('non-existed')).toBe(false)
  expect(isFile(fileURLToPath(import.meta.url))).toBe(true)
})
