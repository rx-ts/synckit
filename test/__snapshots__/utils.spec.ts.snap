// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils encodeImportModule 1`] = `"import 'module-name'"`;

exports[`utils encodeImportModule 2`] = `"import './module-name'"`;

exports[`utils encodeImportModule 3`] = `"import globalName from 'module-name';if(!globalThis.globalName)globalThis.globalName=globalName"`;

exports[`utils encodeImportModule 4`] = `"import {named} from 'module-name';if(!globalThis.globalName)globalThis.globalName=named"`;

exports[`utils encodeImportModule 5`] = `"import * as globalName from 'module-name';if(!globalThis.globalName)globalThis.globalName=globalName"`;

exports[`utils generateGlobals 1`] = `"import fetch from 'node-fetch';if(!globalThis.fetch)globalThis.fetch=fetch;import {performance} from 'node:perf_hooks';if(!globalThis.performance)globalThis.performance=performance"`;

exports[`utils generateGlobals 2`] = `"const fetch=require('node-fetch');if(!globalThis.fetch)globalThis.fetch=fetch;const {performance}=require('node:perf_hooks');if(!globalThis.performance)globalThis.performance=performance"`;

exports[`utils generateGlobals 3`] = `"import fetch from 'node-fetch';globalThis.fetch=fetch;import {performance} from 'node:perf_hooks';if(!globalThis.performance)globalThis.performance=performance"`;

exports[`utils generateGlobals 4`] = `"const fetch=require('node-fetch');globalThis.fetch=fetch;const {performance}=require('node:perf_hooks');if(!globalThis.performance)globalThis.performance=performance"`;
